src/RetryingCallableBuilder.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Tobion\Retry;
+
+use Tobion\Retry\ExceptionHandler\DelayMilliseconds;
+use Tobion\Retry\ExceptionHandler\DelegatingStack;
+use Tobion\Retry\ExceptionHandler\IgnoreSpecificExceptions;
+use Tobion\Retry\ExceptionHandler\IgnoreUntilMaxReached;
+
+/**
+ * Builder for configuring the retry logic.
+ *
+ * @author Tobias Schultze <http://tobion.de>
+ */
+class RetryingCallableBuilder
```

Comment:
probably the only class that does not need to be final as people could extend and add custom exception handlers